saleor/graphql/warehouse/mutations.py

```diff
@@ -197,8 +205,27 @@ def perform_mutation(cls, _root, info, **data):
         node_id = data.get("id")
         model_type = cls.get_type_for_model()
         instance = cls.get_node_or_error(info, node_id, only_type=model_type)
+
+        if instance:
+            cls.clean_instance(info, instance)
+
         stocks = (stock for stock in instance.stock_set.only("product_variant"))
-        result = super(WarehouseDelete, cls).perform_mutation(_root, info, **data)
+        address_id = instance.address_id
+        address = instance.address
+
+        db_id = instance.id
+        instance.delete()
+
+        instance.id = db_id
+        address.id = address_id
+        instance.address = address
+        instance.from_delete_view = True
```

Comment:
Maybe `from_delete_mutation` to be more precise? Also, I would add a comment above these lines to explain why we're assigning the address like this (as this is a bit unusual).